source/databricks/calculation_engine/tests/calculation/energy/test_energy_results.py

```diff
@@ -51,109 +48,99 @@ def factory(
-                ignore_index=True,
-            ),
-            schema=energy_results_schema,
-        ).withColumn(
-            Colname.time_window,
-            struct(
-                col(Colname.time_window).getItem(Colname.start).alias(Colname.start),
-                col(Colname.time_window).getItem(Colname.end).alias(Colname.end),
-            ),
-        )
+        row = {
+            Colname.grid_area: grid_area,
+            Colname.to_grid_area: to_grid_area,
+            Colname.from_grid_area: from_grid_area,
+            Colname.balance_responsible_id: balance_responsible_id,
+            Colname.energy_supplier_id: energy_supplier_id,
+            Colname.time_window: {
+                Colname.start: time_window_start,
+                Colname.end: time_window_end,
+            },
+            Colname.sum_quantity: sum_quantity,
+            Colname.qualities: [quality],
+            Colname.metering_point_type: metering_point_type,
+            Colname.settlement_method: settlement_method,
+        }
+
+        return spark.createDataFrame([Row(**row)], schema=energy_results_schema)
 
     return factory
 
 
-def test__create_dataframe_from_aggregation_result_schema__can_create_a_dataframe_that_match_aggregation_result_schema(
-    input_agg_result_factory,
-) -> None:
-    # Arrange
-    result = input_agg_result_factory()
-
-    # Act
-    actual = EnergyResults(result)
-
-    # Assert
-    assert_schema(
-        actual.df.schema,
-        energy_results_schema,
-        ignore_nullability=True,
-        ignore_decimal_precision=True,
-        ignore_decimal_scale=True,
-    )
-
-
-def test__create_dataframe_from_aggregation_result_schema__match_expected_dataframe(
-    input_agg_result_factory,
-    aggregation_result_factory: Callable[..., DataFrame],
-) -> None:
-    # Arrange
-    result = input_agg_result_factory()
-    expected = aggregation_result_factory(
-        grid_area="A",
-        time_window_start=datetime(2020, 1, 1, 0, 0),
-        time_window_end=datetime(2020, 1, 1, 1, 0),
-        sum_quantity=Decimal("1.234"),
-        quality=QuantityQuality.ESTIMATED.value,
-        metering_point_type=MeteringPointType.CONSUMPTION.value,
-    )
-    # Act
-    actual = EnergyResults(result)
-    # Assert
-    assert actual.df.collect() == expected.collect()
+class TestCtor:
+    class TestWhenNullableColumnsAreMissingInInputDataframe:
+        def test_returns_dataframe_that_includes_missing_column(
+            self,
+            dataframe_with_energy_result_schema_factory,
+        ) -> None:
+            # Arrange
+            df = dataframe_with_energy_result_schema_factory()
+            nullable_columns = [
+                col_name for col_name, nullable in df.dtypes if nullable == "true"
+            ]
+            df_with_missing_columns = df.drop(*nullable_columns)
+
+            # Act
+            actual = EnergyResults(df_with_missing_columns)
+
+            # Assert
+            assert_schema(
```

Comment:
Observation: This seems to me as an implicit way of asserting that the columns are present. I'd prefer more direct and explicit. Perhaps something like:
```suggestion
            assert actual.df.schema.fields[Colname.energy_supplier_id] is not None
```

Reply:
I have changed the test to do something close to that

Reply:
Now, I have explicitly added all the expected nullable columns